### PR TITLE
Implement retention

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
This PR partially fixes https://github.com/Cap-go/capgo/issues/270 and it implements https://github.com/Cap-go/capgo/issues/257. I have added a `retention` column in the `apps` table that defaults to `2592000`. 

I also added a cron job to remove any version that is older then retention as mentioned in the issue.
I also fixed the webhooks from not firing. The wierd IP was found in some supabase issue, but I no longer have the link to that.
I tested all of my changes. Here is a video of these changes in action:

https://github.com/Cap-go/capgo/assets/50914789/f2933805-bf30-4925-a8ed-2fdb4ca6f70f

There is a seperate PR [here](https://github.com/Cap-go/CLI/pull/120) in the CLI to set retention from the CLI

/claim #257

/claim #270

### CODE REVIEW

File: supabase/seed.sql [hunk 1]
```diff
@@ -111,4 +111,15 @@ SELECT http_set_curlopt(''CURLOPT_TIMEOUT_MS'', ''15000'');
            ''application/json'',
            ''{}''
         )::http_request)
-    ', 'localhost', 5432, 'postgres', 'supabase_admin', 't', 'cron_everyday_web_stats');
+    ', 'localhost', 5432, 'postgres', 'supabase_admin', 't', 'cron_everyday_web_stats'),
+(15, '0 0 * * *', '
+    UPDATE app_versions
+    SET deleted = true
+    FROM apps
+    WHERE app_versions.storage_provider=''r2''
```
Comment (riderx) [inline supabase/seed.sql]: any storage is ok, I don't think we need to select only r2

File: supabase/seed.sql [hunk 2]
```diff
@@ -111,4 +111,15 @@ SELECT http_set_curlopt(''CURLOPT_TIMEOUT_MS'', ''15000'');
...
+    AND app_versions.id not in (select app_versions.id from app_versions join channels on app_versions.id = channels.version)
+    AND app_versions.deleted = false
+    AND apps.retention > 0
+    AND extract(epoch from now()) - extract(epoch from app_versions.created_at) > apps.retention
+    AND extract(epoch from now()) - extract(epoch from app_versions.updated_at) > apps.retention
```
Comment (riderx) [inline supabase/seed.sql]: I think the updated_at check should be on app_version_meta, that where the stats are stored, and stats are produce on active bundles

**Comment by riderx** [comment]
Thanks a lot, that a very good solution you made !
I like the cron job only in SQL.
I think that the best solution, after talking about my comments

File: supabase/seed.sql (hunk 1 above)
Comment (WcaleNieWolny) [reply supabase/seed.sql]: Fixed in latest commit

File: supabase/seed.sql (hunk 2 above)
Comment (WcaleNieWolny) [reply supabase/seed.sql]: Would you prefer
```
WHERE app_versions_meta.id = app_versions.id
``` 
or
```
WHERE app_versions_meta.app_id = app_versions.app_id
```
(both work)

Comment (WcaleNieWolny) [reply supabase/seed.sql]: I prefer the second option as it is less likely to break

Comment (riderx) [reply supabase/seed.sql]: agree

**Comment by WcaleNieWolny** [comment]
@riderx Resolved all of the issues mentioned by you

**Comment by riderx** [comment]
I think we have tiny issue the CLI Pr say the retention is set in days (what is logical for me)
but the cron task look for seconds right ?

**Comment by WcaleNieWolny** [comment]
That is true, timestamp diff is in seconds. I did not checked my PR with not my CLI changes

**Comment by riderx** [comment]
I just modified the CLI code to received days and transform it in sec

**Pull request merged**